Engine/src/Console/CommandsHistory.bf

```diff
@@ -0,0 +1,75 @@
+using System;
+
+namespace SteelEngine.Console
+{
+	class CommandsHistory
+	{
+		private String[] _history ~  DeleteContainerAndItems!(_);
+		private int _count = 0;
+		private int _addPos = 0;
+
+		private int _historyPos = 0;
+
+		public this(int maxCount)
+		{
+			_history = new String[maxCount];
+			for (var item in ref _history)
+				item = new String();
+		}
+
+		public void Add(StringView line)
+		{
+			_addPos++;
+			_count++;
+			if (_addPos == _history.Count)
+				_addPos = 0;
+
+			if (_count > _history.Count)
+				_count--;
+
+			let item = _history[_addPos];
+			item.Set(line);
```

Comment:
If a list is used, the history can still wrap around like this, while also allowing a dynamic change in how big the history should be at runtime.